samples/apps/autogen-studio/autogenstudio/workflowmanager.py

```diff
@@ -105,12 +180,22 @@ def _run_workflow(self, message: str, history: Optional[List[Message]] = None, c
         """
         for agent in self.workflow.get("agents", []):
             if agent.get("link").get("agent_type") == "sender":
-                self.sender = self.load(agent.get("agent"))
+                sender_configs = agent.get("agent")
+                self.sender = self.load(sender_configs)
             elif agent.get("link").get("agent_type") == "receiver":
-                self.receiver = self.load(agent.get("agent"))
+                receiver_configs = agent.get("agent")
+                self.receiver = self.load(receiver_configs)
+
         if self.sender and self.receiver:
             # save all agent skills to skills.py
             save_skills_to_file(self.workflow_skills, self.work_dir)
+
+            if self.use_tool_calls:
+                if sender_configs and receiver_configs:
+                    self._connect_tools(sender_configs, receiver_configs)
+                else:
+                    raise ValueError("parameter 'use_tool_calls' invalid with hard-coded Agents")
```

Comment:
Ooof...  another experiment I was doing.  Never mind the "hard coded agents" message.  I'll get that out of there.